### PR TITLE
CLOUDP-205691: update evergreen version

### PR DESCRIPTION
**Pull request by wtrocki**
Change the version on evergreen to remove the patch. This will remove need for us to do numerous code bumps.

### CODE REVIEW

File: build/ci/release.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 variables:
   - &go_linux_version
-      go_root: "/opt/golang/go1.21.3"
-      go_bin: "/opt/golang/go1.21.3/bin"
+      go_root: "/opt/golang/go1.21"
+      go_bin: "/opt/golang/go1.21/bin"
       go_base_path: ""
   - &go_windows_version
       go_root: "c:\\golang\\go1.19"
```
Comment (gssbzn) [inline build/ci/release.yml:7]: seems like windows is still using go 1.19

Comment (wtrocki) [reply build/ci/release.yml:7]: I forgot to ask you. In the last bump we did not moved it to 1.20?:
https://github.com/mongodb/mongodb-atlas-cli/pull/2085/files

Was there any reason for it?

Comment (gssbzn) [reply build/ci/release.yml:7]: probably overlooked

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**